dynamodb_feature_store.js

```diff
@@ -77,10 +77,9 @@ function dynamoDBFeatureStoreInternal(tableName, options) {
         // Write all initial data (without version checks).
         var ops = [];
         allData.forEach(function(collection) {
-          var kindNamespace = collection.kind.namespace;
           collection.items.forEach(function(item) {
             var key = item.key;
-            delete existingNamespaceKeys[kindNamespace + '$' + key];
+            delete existingNamespaceKeys[namespaceForKind(collection.kind) + '$' + key];
```

Comment:
This was the cause of the "duplicate keys" error. If the prefix was, say, `"prefix"`, and we had a flag called `f`, then the logic on lines 70-75 would populate `existingNamespaceKeys` with `"prefix:features$f"`. But then on the line above, `kindNamespace` would just be `features` without the prefix, so we would be trying to remove `"features$f"`. Thus, `"prefix:features$f"` would still be left in the map, so we would later try to delete that key. And since the delete request would probably be in the same batch of operations as the put request for flag `f`, we'd get a duplicate key error.